src/shell/command_line.cpp

```diff
@@ -374,3 +374,74 @@ void CommandLine::Shift(unsigned int amount)
 		}
 	}
 }
+
+bool CommandLine::FindBoolArgument(const std::string& name, bool remove,
+                                   char short_letter)
+{
+	const std::string double_dash = "--" + name;
+	const std::string dash        = '-' + name;
+	char short_name[3]            = {};
+	short_name[0]                 = '-';
+	short_name[1]                 = short_letter;
+	return FindExist(double_dash.c_str(), remove) ||
+	       FindExist(dash.c_str(), remove) ||
+	       (short_letter && FindExist(short_name, true));
+}
+
+std::string CommandLine::StringArgHelper(const char* name)
+{
+	cmd_it it;
+	while (FindEntry(name, it, true)) {
+		cmd_it it_next = it;
+		++it_next;
+		std::string value = *it_next;
+		bool is_valid     = !value.empty() && value[0] != '-';
+		if (is_valid) {
+			++it_next;
+		}
+		cmds.erase(it, it_next);
+		if (is_valid) {
+			return value;
+		}
+	}
+	return {};
+}
+
+std::string CommandLine::FindRemoveStringArgument(const std::string& name)
+{
+	const std::string double_dash = "--" + name;
+	const std::string dash        = '-' + name;
+
+	std::string ret = StringArgHelper(double_dash.c_str());
+	if (ret.empty()) {
+		ret = StringArgHelper(dash.c_str());
+	}
+	return ret;
+}
+
+std::vector<std::string> CommandLine::FindRemoveVectorArgument(const std::string& name)
+{
+	std::vector<std::string> ret = {};
+	while (1) {
```

Comment:
Minor: i `for (;;)` is pretty standard, or even `while (true)`. Kinda dislike there's a number there... weird, i know 😄